### PR TITLE
cookies: session tracking opt in feature

### PR DESCRIPTION
**Pull request by apokellypse**
I set it to true by default. I think we want some way for the client to customize it further, but it didn't occur to me how we wanted to do this, so decided to put up a patch for now.

### CODE REVIEW

**Comment by apokellypse** [comment]
Sorry, I investigated and realized I misunderstood the prompt. Closing this for now.

**Comment by apokellypse** [comment]
After some discussion, It turns out that this approach is on the right track, and the client should be able to set the Session opt in, similar to how conversion tracking is done. Reopening.

**Comment by data-enabler** [comment]
Looks like my commits are still being included in this? You can always just rebase locally and `push -f` if you need to.

**Comment by apokellypse** [comment]
Got rid of the commit sandwich! Ended up using cherry-picking after interactive rebase didn't handle my merge commit and initial commit well.

File: src/answers-umd.js [hunk 1]
```diff
@@ -140,6 +140,7 @@ class Answers {
     });
     globalStorage.setAll(persistentStorage.getAll());
     globalStorage.set(StorageKeys.SEARCH_CONFIG, config.search);
+    globalStorage.set(StorageKeys.SESSIONS_OPT_IN, config.sessionTrackingEnabled);
```
Comment (data-enabler) [inline src/answers-umd.js]: If we're adding a config option, we should add it to the table in the readme.

File: src/core/services/searchservice.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ export default class SearchService {
    * @param {boolean} query.isDynamicFiltersEnabled If true, asks the server to return dynamic filters
    * @param {string} query.skipSpellCheck The boolean as string to indicate if it should skip spell checking
    * @param {string} query.queryTrigger The source that triggers query such as suggest
+   * @param {string} query.sessionTrackingEnabled The boolean as string to indicate if session tracking is on
```
Comment (data-enabler) [inline src/core/services/searchservice.js]: Seems weird to require users of the interface to deal with stringifying the boolean, and we already have a boolean param in `isDynamicFiltersEnabled`.

Comment (apokellypse) [reply src/core/services/searchservice.js]: good point, will make this a boolean

File: src/answers-umd.js (hunk 1 above)
Comment (apokellypse) [reply src/answers-umd.js]: will do

File: src/answers-umd.js [hunk 3]
```diff
@@ -140,6 +140,8 @@ class Answers {
     });
     globalStorage.setAll(persistentStorage.getAll());
     globalStorage.set(StorageKeys.SEARCH_CONFIG, config.search);
+    let sessionTrackingEnabled = config.sessionTrackingEnabled || true;
```
Comment (data-enabler) [inline src/answers-umd.js]: Doesn't this always evaluate to true?

**Review by data-enabler — CHANGES_REQUESTED** [review]

File: src/answers-umd.js (hunk 3 above)
Comment (apokellypse) [reply src/answers-umd.js]: Yes, will fix

**Review by data-enabler — APPROVED** [review]

**Pull request merged**